app/models/levelling_up_premium_payments/award.rb

```diff
@@ -1,42 +1,2710 @@
 module LevellingUpPremiumPayments
-  # This is concerned with the award information from the spreadsheet, which will
-  # contain the pre-computed amount, thereby we don't need to check the EIA or pupil premium
-  # status ourselves.
-  #
-  # The only reason for this class to change is regarding how the monetary amount
-  # is stored.
   class Award
-    # The spreadsheet contains the pre-computed amount as a currency string
-    # like "£2,000.00" or "No payment" and is attached to
-    # https://dfedigital.atlassian.net/browse/CAPT-253
-    #
-    # We'll need a job to perform the initial database storage of this data.
-    def initialize(school)
+    # The spreadsheet attached to https://dfedigital.atlassian.net/browse/CAPT-253
+    # contains the pre-computed amount as a currency string like "£2,000.00" or
+    # "No payment".
+    def initialize(school:, year:)
       raise "nil school" if school.nil?
+      raise "nil year" if year.nil?
+
+      raise "no LUP award mapping for #{year}" unless year.in? self.class.year_to_awards
 
       @school = school
+      @year = year
+    end
+
+    def self.max(year)
+      if @year_to_max.nil?
+        @year_to_max = {}
+      end
+
+      unless @year_to_max.has_key?(year)
+        @year_to_max.store(year, urn_to_award_amount_in_pounds(year).values.max)
+      end
+
+      @year_to_max.fetch(year)
+    end
+
+    def self.urn_to_award_amount_in_pounds(year)
+      year_to_awards.fetch(year)
     end
 
-    def self.max
-      @@maximum_award_amount_in_pounds
+    def self.year_to_awards
+      {AcademicYear.new(2022) => urn_to_award_amount_in_pounds_for_2022_to_2023}
     end
 
     def has_award?
-      amount_in_pounds.positive?
+      check_for_positive_value_in_case_zero_award_has_crept_into_hash
     end
 
     def amount_in_pounds
-      @@urn_to_award_amount_in_pounds.fetch(@school.urn, 0)
+      self.class.urn_to_award_amount_in_pounds(@year).fetch(@school.urn, 0)
+    end
+
+    private_class_method def self.urn_to_award_amount_in_pounds_for_2022_to_2023
+      @urn_to_award_amount_in_pounds_for_2022_to_2023 ||= {
```

Comment:
Why memoize here?

Reply:
My thinking was because the hash is huge and I thought it would avoid repeated hash creation but you might be right. Also it's a class method so there's not multiple copies of this massive hash.